packages/core/ios/Sources/DdSdk.swift

```diff
@@ -5,46 +5,204 @@
+        for (key, value) in castedAttributes {
+            Global.rum.addAttribute(forKey: key, value: value)
+            GlobalState.addAttribute(forKey: key, value: value)
+        }
+        
         resolve(nil)
     }
 
     @objc(setUser:withResolver:withRejecter:)
     func setUser(user: NSDictionary, resolve:RCTPromiseResolveBlock, reject:RCTPromiseRejectBlock) -> Void {
-        nativeInstance.setUser(user: user)
+        var castedUser = castAttributesToSwift(user)
+        let id = castedUser.removeValue(forKey: "id") as? String
+        let name = castedUser.removeValue(forKey: "name") as? String
+        let email = castedUser.removeValue(forKey: "email") as? String
+        let extraInfo: [String: Encodable] = castedUser // everything what's left is an `extraInfo`
+
+        Datadog.setUserInfo(id: id, name: name, email: email, extraInfo: extraInfo)
         resolve(nil)
     }
 
     @objc(setTrackingConsent:withResolver:withRejecter:)
     func setTrackingConsent(trackingConsent: NSString, resolve:RCTPromiseResolveBlock, reject:RCTPromiseRejectBlock) -> Void {
-        nativeInstance.setTrackingConsent(trackingConsent: trackingConsent)
+        Datadog.set(trackingConsent: buildTrackingConsent(consent: trackingConsent))
         resolve(nil)
     }
 
+    private func buildConfiguration(configuration: DdSdkConfiguration) -> Datadog.Configuration {
+        let ddConfigBuilder: Datadog.Configuration.Builder
+        if let rumAppID = configuration.applicationId as String? {
+            ddConfigBuilder = Datadog.Configuration.builderUsing(
+                rumApplicationID: rumAppID,
+                clientToken: configuration.clientToken as String,
+                environment: configuration.env as String
+            )
+            .set(rumSessionsSamplingRate: Float(configuration.sampleRate ?? 100.0))
+        } else {
+            ddConfigBuilder = Datadog.Configuration.builderUsing(
+                clientToken: configuration.clientToken as String,
+                environment: configuration.env as String
+            )
+        }
+
+        switch configuration.site?.lowercased ?? "us" {
+        case "us1", "us":
+            _ = ddConfigBuilder.set(endpoint: .us1)
+        case "eu1", "eu":
+            _ = ddConfigBuilder.set(endpoint: .eu1)
+        case "us3":
+            _ = ddConfigBuilder.set(endpoint: .us3)
+        case "us5":
+            _ = ddConfigBuilder.set(endpoint: .us5)
+        case "us1_fed", "gov":
+            _ = ddConfigBuilder.set(endpoint: .us1_fed)
+        default:
+            _ = ddConfigBuilder.set(endpoint: .us1)
+        }
+
+        let additionalConfig = configuration.additionalConfig
+
+        if let additionalConfiguration = additionalConfig as? [String: Any] {
+            _ = ddConfigBuilder.set(additionalConfiguration: additionalConfiguration)
+        }
+
+        if let enableViewTracking = additionalConfig?["_dd.native_view_tracking"] as? Bool, enableViewTracking {
```

Comment:
nit: It would be nice to avoid hardcoded constants. What might work better is isolating them into single place, with good explanatory comments, similar to what we do with [`CrossPlatformAttributes`](https://github.com/DataDog/dd-sdk-ios/blob/f46f26b6ada8842db224ebb9ff523b3a57a24a3d/Sources/Datadog/Core/Attributes/Attributes.swift#L67-L106) in `dd-sdk-ios`.